Source/Plugin.BLE.Abstractions/DeviceBase.cs

```diff
@@ -53,11 +61,17 @@ protected DeviceBase(IAdapter adapter)
 
         public async Task<IList<IService>> GetServicesAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (!KnownServices.Any())
+            if (KnownServices.Count == 0)
             {
+                IEnumerable<IService> services = null;
                 using (var source = this.GetCombinedSource(cancellationToken))
                 {
-                    KnownServices.AddRange(await GetServicesNativeAsync());
+                    services = await GetServicesNativeAsync();
+                }
+
+                lock (_servicesLock)
+                {
+                    KnownServices.AddRange(services);
```

Comment:
Can `GetServicesNative` return null or timeout? If that's the case, `services` can be null here, so I think we should add a null check

Reply:
Good point. Fixed